artifact/image/image.go

```diff
@@ -158,66 +206,76 @@ func (a Artifact) inspect(ctx context.Context, missingImage string, layerKeys []
 			return xerrors.Errorf("timeout: %w", ctx.Err())
 		}
 	}
-
-	if missingImage != "" {
-		log.Logger.Debugf("Missing image cache: %s", missingImage)
-		if err := a.inspectConfig(missingImage, osFound); err != nil {
-			return xerrors.Errorf("unable to analyze config: %w", err)
+	for _, cache := range a.caches {
+		if missingImage != "" && cache.Type() == types.BuiltInCache {
+			log.Logger.Debugf("Missing image cache: %s", missingImage)
+			if err := a.inspectConfig(missingImage, osFound, cache); err != nil {
+				return xerrors.Errorf("unable to analyze config: %w", err)
+			}
 		}
 	}
 
 	return nil
 
 }
 
-func (a Artifact) inspectLayer(ctx context.Context, diffID string) (types.BlobInfo, error) {
+func (a Artifact) inspectLayer(ctx context.Context, diffID string) (layerInfo map[types.CacheType]types.BlobInfo, err error) {
 	log.Logger.Debugf("Missing diff ID: %s", diffID)
-
+	layerInfo = map[types.CacheType]types.BlobInfo{}
 	layerDigest, r, err := a.uncompressedLayer(diffID)
 	if err != nil {
-		return types.BlobInfo{}, xerrors.Errorf("unable to get uncompressed layer %s: %w", diffID, err)
+		return layerInfo, xerrors.Errorf("unable to get uncompressed layer %s: %w", diffID, err)
 	}
 
 	// below line of code gets the size of uncompressed layer. Will sum up these layer sizes to get the size of image.
 	cr := newCountingReader(r)
 	var wg sync.WaitGroup
-	result := new(analyzer.AnalysisResult)
+	var resultMap = map[types.CacheType]*analyzer.AnalysisResult{}
+	for _, cache := range a.caches {
+		resultMap[cache.Type()] = new(analyzer.AnalysisResult)
+	}
 	limit := semaphore.NewWeighted(parallel)
-
 	opqDirs, whFiles, err := walker.WalkLayerTar(cr, func(filePath string, info os.FileInfo, opener analyzer.Opener) error {
-		if err = a.analyzer.AnalyzeFile(ctx, &wg, limit, result, filePath, info, opener); err != nil {
+		if err = a.analyzer.AnalyzeFile(ctx, &wg, limit, resultMap, filePath, info, opener); err != nil {
 			return xerrors.Errorf("failed to analyze %s: %w", filePath, err)
 		}
 		return nil
 	})
 	if err != nil {
-		return types.BlobInfo{}, xerrors.Errorf("walk error: %w", err)
+		return layerInfo, xerrors.Errorf("walk error: %w", err)
 	}
 
 	// Wait for all the goroutine to finish.
 	wg.Wait()
 
-	// Sort the analysis result for consistent results
-	result.Sort()
+	for cache, _ := range resultMap {
```

Comment:
Any reason we cannot use the second argument?

```suggestion
	for cache, result := range resultMap {
```